### PR TITLE
Dockerfile: Update CLI, buildx and compose

### PR DESCRIPTION
**Pull request by vvoland**
### Dockerfile: update docker CLI to v27.2.1

Update the Docker CLI used in the dev-container

full diff: https://github.com/docker/cli/compare/v27.0.2...v27.2.1

### Dockerfile: update buildx to 0.17.0

Update the buildx cli plugin used in the dev-container

full diff: https://github.com/docker/buildx/compare/0.16.1...0.17.0

### Dockerfile: update compose to v2.29.2

Update the compose cli plugin used in the dev-container

full diff: https://github.com/docker/cli/compare/v2.29.0...v2.29.2

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,11 +8,11 @@ ARG XX_VERSION=1.5.0
...
 
 ARG DOCKERCLI_REPOSITORY="https://github.com/docker/cli.git"
-ARG DOCKERCLI_VERSION=v27.0.2
+ARG DOCKERCLI_VERSION=v27.2.1
 # cli version used for integration-cli tests
 ARG DOCKERCLI_INTEGRATION_REPOSITORY="https://github.com/docker/cli.git"
 ARG DOCKERCLI_INTEGRATION_VERSION=v17.06.2-ce
-ARG BUILDX_VERSION=0.16.1
+ARG BUILDX_VERSION=0.17.0
 ARG COMPOSE_VERSION=v2.29.0
```
Comment (thaJeztah) [inline Dockerfile:16]: Compose can also be updated to v2.29.2; https://github.com/docker/compose/releases/tag/v2.29.2

**Review by thaJeztah — APPROVED** [review]
LGTM

left a suggestion for compose

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by crazy-max — APPROVED** [review]

**Review by laurazard — APPROVED** [review]

**Comment by thaJeztah** [comment]
Failure is unrelated and tracked in https://github.com/moby/moby/issues/48400

**Pull request merged**